azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,978 @@
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"operation": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"Scale"}, true),
+						},
+						"email": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"send_to_subscription_administrator": {
+										Type:     schema.TypeBool,
+										Required: true,
+									},
+									"send_to_subscription_co_administrator": {
+										Type:     schema.TypeBool,
+										Required: true,
+									},
+									"custom_emails": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+								},
+							},
+							Set: resourceAzureRmAutoscaleDefaultHash,
+						},
+						"webhook": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							MinItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"service_uri": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"properties": {
+										Type:     schema.TypeMap,
+										Optional: true,
+									},
+								},
+							},
+							Set: resourceAzureRmAutoscaleDefaultHash,
+						},
+					},
+				},
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"target_resource_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmAutoscaleSettingCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	armClient := meta.(*ArmClient)
+	asClient := armClient.autoscaleSettingsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroupName := d.Get("resource_group_name").(string)
+	location := d.Get("location").(string)
+	enabled := d.Get("enabled").(bool)
+	targetResourceURI := d.Get("target_resource_id").(string)
+	tags := d.Get("tags").(map[string]interface{})
+	expandedTags := expandTags(tags)
+
+	profiles, err := expandAzureRmAutoscaleProfile(d)
+	if err != nil {
+		return err
+	}
+
+	notifications, err := expandAzureRmAutoscaleNotification(d)
+	if err != nil {
+		return err
+	}
+
+	autoscaleSetting := insights.AutoscaleSetting{
+		Name:              &name,
+		Enabled:           &enabled,
+		TargetResourceURI: &targetResourceURI,
+		Profiles:          &profiles,
+		Notifications:     &notifications,
+	}
+
+	parameters := insights.AutoscaleSettingResource{
+		Name:             &name,
+		Location:         &location,
+		Tags:             expandedTags,
+		AutoscaleSetting: &autoscaleSetting,
+	}
+
+	result, err := asClient.CreateOrUpdate(ctx, resourceGroupName, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*result.ID)
```

Comment:
The response `ID` of `CreateOrUpdate` may not be correct (especially the casing), so we are still required to use the `Get` method to read the correct `ID` just like [other resources](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_redis_cache.go#L265).